src/game/client/components/scoreboard.cpp

```diff
@@ -103,50 +142,117 @@ void CScoreboard::RenderSpectators(float x, float y, float w)
 
 		if(Multiple)
 			TextRender()->TextEx(&Cursor, ", ", -1);
-		if(pInfo->m_PlayerFlags&PLAYERFLAG_WATCHING)
-			TextRender()->TextColor(1.0f, 1.0f, 0.0f, 1.0f);
-		char aBuf[64];
-		str_format(aBuf, sizeof(aBuf), "%2d: %s", i, g_Config.m_ClShowsocial ? m_pClient->m_aClients[i].m_aName : "");
-		TextRender()->TextEx(&Cursor, aBuf, -1);
+		if(m_pClient->m_aClients[i].m_aClan[0])
+		{
+			str_format(aBuf, sizeof(aBuf), "%s ", m_pClient->m_aClients[i].m_aClan);
+			TextRender()->TextColor(1.0f, 1.0f, (pInfo->m_PlayerFlags&PLAYERFLAG_WATCHING) ? 0.0f : 1.0f, 0.7f);
+			TextRender()->TextEx(&Cursor, aBuf, -1);
+		}
+		TextRender()->TextColor(1.0f, 1.0f, (pInfo->m_PlayerFlags&PLAYERFLAG_WATCHING) ? 0.0f :	 1.0f, 1.0f);
+		TextRender()->TextEx(&Cursor, m_pClient->m_aClients[i].m_aName, -1);
 		TextRender()->TextColor(1.0f, 1.0f, 1.0f, 1.0f);
 		Multiple = true;
 	}
+
+	return RectHeight;
 }
 
-void CScoreboard::RenderScoreboard(float x, float y, float w, int Team, const char *pTitle)
+float CScoreboard::RenderScoreboard(float x, float y, float w, int Team, const char *pTitle, int Align)
 {
 	if(Team == TEAM_SPECTATORS)
-		return;
+		return 0.0f;
 
-	float h = 760.0f;
-	CUIRect Rect = {x, y, w, h};
+	float HeadlineHeight = 40.0f;
+	float TitleFontsize = 20.0f;
+	float HeadlineFontsize = 12.0f;
+	float LineHeight = 20.0f;
+	float TeeSizeMod = 1.0f;
+	float Spacing = 2.0f;
+	float NameOffset = x+4.0f, NameLength = 136.0f;
+	float TeeOffset = x+4.0f, TeeLength = 25*TeeSizeMod;
+	float ClanOffset = NameOffset+NameLength, ClanLength = 90.0f;
+	float KillOffset = ClanOffset+ClanLength, KillLength = 30.0f;
+	float DeathOffset = KillOffset+KillLength, DeathLength = 30.0f;
+	float ScoreOffset = DeathOffset+DeathLength, ScoreLength = 35.0f;
+	float PingOffset = ScoreOffset+ScoreLength, PingLength = 35.0f;
+	float tw = 0.0f;
+
+	bool NoTitle = pTitle? false : true;
+
+	// count players
```

Comment:
No need for calculation: 'm_pClient->m_GameInfo.m_aTeamSize[Team]' has it stored, but would add a check on top to leave the function if the team isn't red or blue.

Reply:
I added an assert instead. There really should be no other team.

Reply:
I would add an assert instead, there really shouldn't be any other team.